api/posts.go

```diff
@@ -80,6 +93,55 @@ func GetPosts(c *cli.Context) *[]Post {
 	return &posts
 }
 
+func GetUserPosts(c *cli.Context) ([]RemotePost, error) {
+	waposts, err := DoFetchPosts(c)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(waposts) == 0 {
+		return nil, nil
+	}
+
+	posts := []RemotePost{}
+	for _, p := range waposts {
+		post := RemotePost{
+			Post: Post{
+				ID:        p.ID,
+				EditToken: p.Token,
+			},
+			Title:   p.Title,
+			Excerpt: getExcerpt(p.Content),
+			Slug:    p.Slug,
+			Updated: p.Updated,
+		}
+		if p.Collection != nil {
+			post.Collection = p.Collection.Alias
+		}
+		posts = append(posts, post)
+	}
+
+	return posts, nil
+}
+
+// getExcerpt takes in a content string and returns
+// a concatenated version. limited to no more than
+// two lines of 80 chars each. delimited by '...'
+func getExcerpt(input string) string {
+	length := len(input)
+
+	if length <= 80 {
+		return input
+	}
+
+	if length <= 160 {
+		return input[:80] + "\n" + input[80:]
```

Comment:
Note this could break multi-byte characters. We should do these operations on a `[]rune` instead of on the `string`, like we do in [`parse.TruncToWord()`](https://github.com/writeas/writefreely/blob/98f5b1489900c5b9957426e2bb03b3c9ed98d8d8/parse/posts.go#L65).

Reply:
I was thinking about that, good call. My brain was getting mushy yesterday.

Reply:
No worries!